src/v8/inspector/mod.rs

```diff
@@ -0,0 +1,425 @@
+        }
+    }
+
+    /// Sets the callback which is used by the debugger to send messages
+    /// to the remote client.
+    fn set_on_response_callback(
+        raw: &RawInspector,
+        on_response_callback: Box<OnResponseCallback>,
+    ) -> Box<Box<OnResponseCallback>> {
+        let on_response_callback = Box::new(on_response_callback);
+        let on_response_callback = Box::into_raw(on_response_callback);
+
+        unsafe {
+            crate::v8_c_raw::bindings::v8_InspectorSetOnResponseCallback(
+                raw.raw.as_ptr(),
+                Some(on_response),
+                on_response_callback as _,
+            );
+        }
+
+        unsafe { Box::from_raw(on_response_callback as *mut _) }
+    }
+
+    /// Sets the callback when the debugger needs to wait for the
+    /// remote client's message.
+    fn set_on_wait_frontend_message_on_pause_callback(
+        raw: &RawInspector,
+        on_wait_frontend_message_on_pause_callback: Box<OnWaitFrontendMessageOnPauseCallback>,
+    ) -> Box<Box<OnWaitFrontendMessageOnPauseCallback>> {
+        let on_wait_frontend_message_on_pause_callback =
+            Box::new(on_wait_frontend_message_on_pause_callback);
+        let on_wait_frontend_message_on_pause_callback =
+            Box::into_raw(on_wait_frontend_message_on_pause_callback);
+
+        unsafe {
+            crate::v8_c_raw::bindings::v8_InspectorSetOnWaitFrontendMessageOnPauseCallback(
+                raw.raw.as_ptr(),
+                Some(on_wait_frontend_message_on_pause),
+                on_wait_frontend_message_on_pause_callback as _,
+            );
+        }
+
+        unsafe { Box::from_raw(on_wait_frontend_message_on_pause_callback as *mut _) }
+    }
+}
+
+impl Deref for Inspector {
+    type Target = RawInspector;
+
+    fn deref(&self) -> &Self::Target {
+        &self.raw
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::v8::*;
+
+    #[test]
+    fn set_breakpoint() {
+        // Initialise the V8 engine:
+        crate::test_utils::initialize();
+
+        // Create a new isolate:
+        let isolate = isolate::V8Isolate::new();
+
+        // Enter the isolate created:
+        let i_scope = isolate.enter();
+
+        // Create a JS execution context for code invocation:""
+        let ctx = i_scope.new_context(None);
+
+        // Enter the created execution context for debugging:
+        let ctx_scope = ctx.enter(&i_scope);
+
+        // Create an inspector.
+        let inspector = RawInspector::new(&ctx_scope);
+
+        // Set a "good" breakpoint.
+        assert!(inspector
+            .schedule_pause_on_next_statement("Test breakpoint")
+            .is_ok());
+
+        // Set a "bad" breakpoint.
+        assert!(inspector
+            .schedule_pause_on_next_statement("Test\0breakpoint")
+            .is_err());
+    }
+
+    // TODO: Unfortunately, this test crashes the V8 engine, it requires
```

Comment:
We need to not forget about this crash, we need to make sure we are not doing something wrong.

Reply:
What about this test? Did we understood why is it crashing?

Reply:
OK, so lets just keep it commented to remember that we need to check it, either find out that we have an issue or maybe open an issue to V8.